editor/test/integration/save_data_test.clj

```diff
@@ -0,0 +1,1372 @@
+  :sanitize-fn usage in non-component resource types."
+  (testing "The project should not have unsaved changes immediately after loading."
+    (check-project-save-data-disk-equivalence! project/dirty-save-data)))
+
+(deftest no-unsaved-changes-after-save-test
+  "This test is intended to verify that we track unsaved changes properly. If
+  any other tests in this module are failing as well, you should address them
+  first."
+  (test-util/with-scratch-project project-path
+    (clear-cached-save-data!)
+    (let [checked-resources (checked-resources workspace)
+          dirty-proj-paths (into (sorted-set)
+                                 (map (comp resource/proj-path :resource))
+                                 (project/dirty-save-data project))]
+      (when (is (= #{} dirty-proj-paths)
+                (list-message
+                  (format "Unable to proceed with test due to unsaved changes in the following files immediately after loading the project `editor/%s`:"
+                          project-path)
+                  dirty-proj-paths))
+        (doseq [resource checked-resources]
+          (let [proj-path (resource/proj-path resource)
+                node-id (test-util/resource-node project resource)]
+            (when (testing (format "File `%s` should not have unsaved changes prior to editing." proj-path)
+                    (let [save-data (g/valid-node-value node-id :save-data)]
+                      (if (not (:dirty? save-data))
+                        true
+                        (let [message (str "Unsaved changes detected before editing. This is likely due to an interdependency between resources. You might need to adjust the order resources are edited.\n"
+                                           (save-data-diff-message save-data))]
+                          (is (not (:dirty? save-data)) message)))))
+              (test-util/edit-resource-node! node-id)
+              (testing (format "File `%s` should have unsaved changes after editing." proj-path)
+                (let [save-data (g/valid-node-value node-id :save-data)]
+                  (is (:dirty? save-data)
+                      "No unsaved changes detected after editing. Possibly, `test-util/edit-resource-node!` is not making a meaningful change to the file?"))))))
+        (test-util/save-project! project)
+        (clear-cached-save-data!)
+        (doseq [resource checked-resources]
+          (let [proj-path (resource/proj-path resource)]
+            (testing (format "File `%s` should not have unsaved changes after saving." proj-path)
+              (let [node-id (test-util/resource-node project resource)
+                    save-data (g/valid-node-value node-id :save-data)]
+                (is (not (:dirty? save-data))
+                    "Unsaved changes detected after saving.")
+                (when (:dirty? save-data)
+                  (check-save-data-disk-equivalence! save-data))))))))))
```

Comment:
This test covers the remaining part of what `save-test/save-dirty` used to cover.